TransformationPlayground/playground_frontend/src/app/page.tsx

```diff
@@ -0,0 +1,206 @@
+  const [transformLanguage, setTransformLanguage] = useState<SelectProps.Option>(
+    transformLanguageOptions[0]
+  );
+
+  // Define the request body for GenAI Recommendations
+  interface GetRecommendationBody {
+    input_shape: any;
+    source_version: string;
+    target_version: string;
+    transform_language: string;
+    test_target_url?: string;
+  }
+
+  // Handle Get Recommendation Request
+  const handleGetRecommendation = async () => {
+    setIsLoading(true); // Start visual spinner
+
+    // Parse the input_shape string into JSON
+    const parsedInputShape = JSON.parse(inputShape);
+
+    const payload: GetRecommendationBody = {
+      input_shape: parsedInputShape,
+      source_version: sourceVersion.value!,
+      target_version: targetVersion.value!,
+      transform_language: transformLanguage.value!
+    };
+
+    // Add optional fields to payload
+    if (testTargetUrl && testTargetUrl.length > 0) {
+      payload.test_target_url = testTargetUrl;
+    }
+
+    try {
+      const response = await fetch("http://localhost:8000/transforms/index/", {
+        method: "POST",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify(payload),
+      });
+
+      if (response.ok) {
+        const data = await response.json();
+
+        // Pretty-print JSON for output_shape
+        const prettyOutputShape = JSON.stringify(data.output_shape, null, 4);
+
+        // Join validation_report with new lines
+        const formattedValidationReport = (data.validation_report || []).join("\n\n");
+
+        setTransformLogic(data.transform_logic || "");
+        setOutputShape(prettyOutputShape || "");
+        setValidationReport(formattedValidationReport || "");
+      } else {
+        console.error("Failed to fetch recommendation");
+      }
+    } catch (error) {
+      console.error("Error:", error);
+
+      if (error instanceof SyntaxError) {
+        alert("Invalid JSON: Please check your input and try again.");
+      }
+    } finally {
+      setIsLoading(false); // Stop visual spinner
+    }
+  };
+
+  return (
+    <Grid
+      gridDefinition={[
+        { colspan: 4 }, // Left Column
+        { colspan: 4 }, // Middle Column
+        { colspan: 4 }, // Right Column
+      ]}
+    >
+      {/* User Input Column */}
+      <Container header="User Input Panel">
+        <FormField
+          label="Original JSON Input"
+          description="Paste or edit the JSON blob that represents the Elasticsearch 6.8 Index JSON."
+        >
+          <Textarea
+            value={inputShape}
+            onChange={({ detail }) => setInputShape(detail.value)}
+            placeholder='{"index_name": "my-index", "index_json": { index settings and mappings }}'
+            rows={30}
+          />
+        </FormField>
+      </Container>
+
+      {/* Transformation Column */}
+      <Container header="Transformation Panel">
```

Comment:
This seems like it should be refactored into its own control to reduce overall coupling

Reply:
Would love more details!  But yeah - this whole page is ripe for refactoring.